Environment/Environment.tex

```diff
@@ -764,6 +764,58 @@ \subsection{The object explorer}
 	\label{fig:explorePluggableListMorph}
 \end{figure}
 
+
+\subsection{The instance browser}
+
+Sometimes, you might want to explore what a specific object can do.
+In the class browser, you can look through the methods to learn what they should do, but you can not execute them and see what they actually do.
+In the inspector or explorer, you can see the state of an object and execute methods, but you can not directly see the code corresponding to the methods.
+In case you want both, reading code and evaluating code for a single object, you can use the \emph{instance browser} tool.
+\index{instance browser}
+The instance browser allows you to read the code of all methods that an object can execute, and evaluate any code snippets in the context of this object.
+
+You can open an instance browser from the explorer.
+\dothis{In the explorer above, select the root entry for the \ct{PluggableListMorph}, open the yellow button menu, and choose \menu{browse protocol (p)}.}
+In the new browser (\figref{fig:instanceBrowser}), you can see a list of methods on the top right, which contains all methods the object can execute, gathered from its class and all its superclasses.
+The methods are categorized according to their protocols on the top left.
+At the bottom you see a code pane, just like in an ordinary browser.
+However, this window is more interesting as it still knows about the \ct{PluggableListMorph} for which we opened it.
+Thus, we can evaluate code in the code pane.
+\dothis{In the browser, open the method \ct{fullListSize} by selecting it in the list at the top right.
+Then, navigate to the last code line and \menu{print it} (\short{p}). 
+You should see the result of evaluating the expression right within the code editor.}
+
+\begin{ExecuteSmalltalkScript}
+	SBEScreenshotRecorder writeTo: './figures/instanceBrowser.png' building: [:helper |
+		| lists list browser textField |
+		SystemBrowser defaultOpenBrowser
+			selectSystemCategory: #'Collections-Strings';
+			selectClass: String.
+		lists := (helper foregroundWindow submorphsSatisfying: #isPluggableListMorph)
+			collect: [:alist | alist center x -> alist] as: Dictionary.
+		list := (lists at: lists keys max).
+		browser := InstanceBrowser new openOnObject: list showingSelector: #fullListSize.
+		browser selectSelectorItsNaturalCategory: #fullListSize. "To work around a defect in *showingSelector:"
+
+		textField := helper codePaneOfBrowser: helper foregroundWindow.
+		helper click: textField.
+		helper
+			keyStroke: textField key: $a modifiers: #(cmd);
+			type: 'fullListSize\	"Number of items in the unfiltered list."\\	^ self getFullList size' withCRs into: textField;
+			keyStroke: textField key: $p modifiers: #(cmd).
+		helper takeScreenshotOf: helper foregroundWindow.

```

Comment:
I think this message send is redundant unless you have `Context>>wrap:` in the script (see `SBEWorldHelper>>#assureScreenshot`)